src/longsequences/transformations.jl

```diff
@@ -11,50 +11,40 @@ array unless the new size does not fit. If `force`, always resize underlying dat
 function Base.resize!(seq::LongSequence{A}, size::Integer, force::Bool=false) where {A}
     if size < 0
         throw(ArgumentError("size must be non-negative"))
-    elseif seq.shared
-        # May seem wasteful to orphan here, but if you resize! a shared seq,
-        # you're probably going to write to it anyway.
-        return _orphan!(seq, size)
     else
         if force | (seq_data_len(A, size) > seq_data_len(A, length(seq)))
             resize!(seq.data, seq_data_len(A, size))
         end
-        seq.part = 1:size
+        seq.len = size
         return seq
     end
 end
 
-function Base.filter!(f::Function, seq::LongSequence{A}) where {A}
-    orphan!(seq)
-
-    len = 0
-    next = bitindex(seq, 1)
-    j = index(next)
-    datum::UInt64 = 0
-    for i in 1:lastindex(seq)
-        x = inbounds_getindex(seq, i)
-        if f(x)
-            datum |= enc64(seq, x) << offset(next)
-            len += 1
-            #TODO: Resolve use of bits_per_symbol.
-            next += bits_per_symbol(A())
-            if index(next) != j
-                seq.data[j] = datum
-                datum = 0
-                j = index(next)
+function Base.filter!(f, seq::LongSequence)
+    writeindex = bitindex(seq, 0)
+    chunk = zero(UInt64)
+    bps = bits_per_symbol(seq)
+    for i in eachindex(seq)
+        encoded_symbol = extract_encoded_element(bitindex(seq, i), encoded_data(seq))
```

Comment:
Doesn't inbounds_getindex do these two lines already?

Reply:
Yep, but I'm using the encoded symbol a few lines down, and don't want to redo the calculation,